### PR TITLE
CS lime-203, Not Accounted Ether Transfers

### PR DESCRIPTION
**Pull request by KalpitaMandal**
# Description
Multiple times the code checks whether the asset address is address zero. If so, it checks if the ether amount equals the provided amount argument `msg.value == amount`. In the case of token operations, it is not checked if Ether was sent in the same transaction. Ether might be sent accidentally in the transaction as the functions are payable. That Ether would not be accounted for.

# Integrations Checklist

- [ ]  Have any function signatures changed? If yes, outline below.
- [ ]  Have any features changed or been added? If yes, outline below.
- [ ]  Have any events changed or been added? If yes, outline below.
- [ ]  Has all documentation been updated?

# Changelog

- Assigning `msg.value` to zero in the following contracts, where ETH transfers are not required:
  - `CreditLine.sol`
  - `Pool.sol`
  - `Repayments.sol`
  - `SavingsAccount.sol`

# Event Signature Changes

# Function Signature Changes

# Features

# Events

### CODE REVIEW

File: contracts/CreditLine/CreditLine.sol [hunk 1]
```diff
@@ -615,6 +615,9 @@ contract CreditLine is ReentrancyGuard, OwnableUpgradeable {
         bool _fromSavingsAccount
     ) external payable nonReentrant ifCreditLineExists(_id) {
         require(creditLineVariables[_id].status == CreditLineStatus.ACTIVE, 'CreditLine not active');
+        if(Address(creditLineConstants[_id].collateralAsset) != Address(0)) {
+            require(msg.value == 0, 'DepositCollateral: ETH is not required for this operation');
```
Comment (prateekreddy) [inline contracts/CreditLine/CreditLine.sol:619]: This can be move to _depositCollateral function for gas optimization

**Comment by prateekreddy** [comment]
Missed  this check in `_repay` function from creditLines

File: contracts/Pool/Pool.sol [hunk 2]
```diff
@@ -172,6 +172,9 @@ contract Pool is Initializable, ERC20PausableUpgradeable, IPool, ReentrancyGuard
     function depositCollateral(uint256 _amount, bool _transferFromSavingsAccount) external payable override {
         require(_amount != 0, 'DC1');
         require(balanceOf(msg.sender) == 0, 'DC2');
+        if(Address(poolConstants.collateralAsset) != Address(0) || _transferFromSavingsAccount) {
```
Comment (prateekreddy) [inline contracts/Pool/Pool.sol:175]: These checks can be move to the savingsAccountUtil functions which will make it more efficient

File: contracts/SavingsAccount/SavingsAccount.sol [hunk 3]
```diff
@@ -112,6 +112,9 @@ contract SavingsAccount is ISavingsAccount, Initializable, OwnableUpgradeable, R
         address _to
     ) external payable override nonReentrant returns (uint256) {
         require(_to != address(0), 'SavingsAccount::deposit receiver address should not be zero address');
+        if(Address(_token) != Address(0)) {
+            require(msg.value == 0, 'Deposit: ETH is not required for this operation');
```
Comment (prateekreddy) [inline contracts/SavingsAccount/SavingsAccount.sol:116]: May  be adding this to depositToYield internal function is more efficient

**Review by prateekreddy — CHANGES_REQUESTED** [review]

**Comment by KalpitaMandal** [comment]
> Missed this check in `_repay` function from creditLines

The check is present in `repay` function which calls the `_repay` function, so the same was not iterated in the internal function. 
Let me know if it i required anyway.

File: contracts/CreditLine/CreditLine.sol (hunk 1 above)
Comment (KalpitaMandal) [reply contracts/CreditLine/CreditLine.sol:619]: The change is made but having the check in the internal function will revert only after another function is called. Putting it in the external function will ensure the internal function is never called if ETH is sent.

File: contracts/Pool/Pool.sol (hunk 2 above)
Comment (KalpitaMandal) [reply contracts/Pool/Pool.sol:175]: Added the change

File: contracts/SavingsAccount/SavingsAccount.sol (hunk 3 above)
Comment (KalpitaMandal) [reply contracts/SavingsAccount/SavingsAccount.sol:116]: Added the change.

**Review by prateekreddy — CHANGES_REQUESTED** [review]
https://github.com/sublime-finance/sublime-v1/blob/CS-lime-203/contracts/SavingsAccount/SavingsAccountUtil.sol#L49 doesn't have `msg.value` check and https://github.com/sublime-finance/sublime-v1/blob/CS-lime-203/contracts/Pool/Pool.sol#L594 function doesn't need to be payable. Receive method can be removed from all contracts. RepayPrinciple in repayments.sol hasn't been covered.

**Review by prateekreddy — APPROVED** [review]

**Pull request merged**